rest/serializers.py

```diff
@@ -0,0 +1,94 @@
+"""
+REST serializers for taxonomy models
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.serializers import (
+    ModelSerializer,
+    HiddenField,
+    CurrentUserDefault,
+    CreateOnlyDefault,
+)
+from rest_framework.generics import get_object_or_404
+
+from taxonomy.models import Vocabulary, Term
+from learningresources.models import Repository
+from .util import LambdaDefault
+
+
+class RepositorySerializer(ModelSerializer):
+    """Serializer for Repository"""
+
+    created_by = HiddenField(
+        default=CreateOnlyDefault(CurrentUserDefault())
+    )
+
+    class Meta:
+        # pylint: disable=missing-docstring
+        model = Repository
+        fields = (
+            'id',
+            'slug',
+            'name',
+            'description',
+            'create_date',
+            'created_by',
+        )
+        read_only_fields = (
+            'id',
+            'slug',
+        )
+
+
+class VocabularySerializer(ModelSerializer):
+    """Serializer for Vocabulary"""
+
+    repository = HiddenField(default=LambdaDefault(
+        lambda context: get_object_or_404(
+            Repository, slug=context['view'].kwargs['repo_slug']
+        )
+    ))
+
+    class Meta:
+        # pylint: disable=missing-docstring
+        model = Vocabulary
+        fields = (
+            'id',
+            'slug',
+            'name',
+            'description',
+            'vocabulary_type',
+            'required',
+            'weight',
+            'repository',
```

Comment:
Okay. So, the `repository` field _should_ be in `fields` because it's useful, and making it hidden tells the serializer to leave it out of the JSON and not accept input values for it on POST?